cypress/e2e/issue_credential_version_1_use_case.cy.ts

```diff
@@ -19,13 +29,24 @@ describe('Onboarding demo test using issue credential protocol version 1', () =>
 
       const oobId = interception.response?.body.outOfBandRecord.id
 
-      cy.intercept('GET', `${API_URL}/connections?outOfBandId=${oobId}`).as('getConnectionRecord')
-
       cy.request('POST', `${TEST_AGENT_URL}/oob/receive-invitation-url`, body)
 
-      cy.wait(['@getConnectionRecord']).then((inter) => {
-        const record = inter.response?.body[0]
-        cy.wrap(record).its('state').should('not.equal', 'invited')
+      cy.streamRequest<Event>(webSocketConfig, {
+        streamTimeout: 10000,
+        // Waits for connection event with oobId and state is completed or response-sent
+        takeWhileFn: (event) => {
+          if (!isConnectionEvent(event)) return true
+
+          return (
+            event.payload.connectionRecord.outOfBandId !== oobId &&
+            ![DidExchangeState.Completed, DidExchangeState.ResponseSent].includes(event.payload.connectionRecord.state)
+          )
+        },
+      }).then((results) => {
+        const length = (results && results.length) || 0
+        const result = results && results[length - 1]
```

Comment:
What do you do here?

Reply:
The websocket package we used delivers the events as an array containing one event. This is just logic to ensure that we're returning the object in the array, and not the array